doc/Makefile

```diff
@@ -68,7 +68,7 @@ mkdir:
 html: $(OUTPUT)
 
 manpage: $(MAN)
-	cat $(VPATH)/man-snippets/header $(MAN) > h2o.8
+	cat $(VPATH)/man-snippets/header $(MAN) > h2o.conf.5
```

Comment:
With this change, to me it seems that h2o.8 is no longer being generated.

Are we expected to modify the file by hand, or as I missing something?

Reply:
> Are we expected to modify the file by hand

That was my intention, yes. We could keep in sync the command line options automatically, but since they don't change much, i thought it wasn't worth it. Does that work for you?

Reply:
> We could keep in sync the command line options automatically, but since they don't change much, i thought it wasn't worth it. Does that work for you?

Thank you for clarifying your intention.

Yes. I am afraid of things getting out-of-sync.

Could we use [help2man](https://www.gnu.org/software/help2man/) to generate the man page for the command? My understanding is that the command is popular at least to some extent, enough to be provided by homebrew.

I see some issues in the man page generated by running `help2man --section=8 --version-string='version' --no-info ./h2o`. But we might be able to address it by changing the output of `h2o --help` (and the added bonus is that doing so will make the output of the help less surprising to humans as well).

Reply:
> Could we use help2man to generate the man page for the command?

What about generating `h2o.8` from `srcdoc/configure/command_options.mt` instead? That way all man pages are generated the same way (from the `.mt` files).

Reply:
That sounds like a good idea!